articles/quickstart/spa/angular-next/02-calling-an-api.md

```diff
@@ -0,0 +1,140 @@
+
+The following is an example of an Angular module (based upon the default implementation when you create a new app using `ng new`) that supports `AuthHttpInterceptor`, configured to call the Auth0 Management API with the ability to read the current user profile:
+
+```javascript
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { AppComponent } from './app.component';
+
+// Import the injector module and the Angular types you'll need
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { AuthHttpInterceptor } from '@auth0/auth0-angular';
+
+@NgModule({
+  declarations: [],
+  imports: [
+    BrowserModule,
+
+    // Register HttpClient for making API calls
+    HttpClientModule,
+    
+    AppRoutingModule,
+    AuthModule.forRoot({
+      domain: "${account.namespace}",
+      clientId: "${account.clientId}",
+
+      // Specify the audience for the management API, allowing us to make API calls
+      audience: "https://${account.namespace}/api/v2/",
+
+      // Specify the necessary scopes for getting user information
+      scope: "read:current_user",
+
+      // Specify configuration for the interceptor              
+      httpInterceptor: {
+        allowedList: [
+          {
+            // Match any request that starts 'https://${account.namespace}/api/v2/' (note the asterisk)
+            uri: 'https://${account.namespace}/api/v2/*'
+            tokenOptions: {
+              // The attached token should target this audience
+              audience: 'https://${account.namespace}/api/v2/',
+
+              // The attached token should have these scopes
+              scope: 'read:current_user'
+            }
+          }
+        ]
+      }
+    }),
+  ],
+  providers: [
+    // Register the `AuthHttpInterceptor` to automatically attach the 'Authorization' header
+    { provide: HTTP_INTERCEPTORS, useClass: AuthHttpInterceptor, multi: true },
+  ],
+  bootstrap: [AppComponent],
+})
+```
+
+## Make an API Call
+
+With your app module configured with the HTTP interceptor from the Angular SDK, calls you make using Angular's built-in `HttpClient` to the Auth0 Management API will have the appropriate access token specified in the `Authorization` header. Let's use this as an example for showing user metadata.
+
+The following component demonstrates how to display the `user_metadata` field from the authenticated user's profile, by making a call to the `/api/v2/users` endpoint using `HttpClient`:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { concatMap, tap } from 'rxjs/operators';
+
+// Import the HttpClient for making API requests
+import { HttpClient } from '@angular/common/http';
+
+// Import AuthService from the Auth0 Angular SDK to get access to the user
+import { AuthService } from '@auth0/auth0-angular';
+
+@Component({
+  selector: 'app-metadata',
+  template: `<div *ngIf="auth.user$ | async as user">
```

Comment:
the template is not using this `user` variable. I guess is better to either use it explicitly or await with ngIf on `metadata` instead.

Reply:
Good spot - I've gone with waiting on metadata.